### PR TITLE
Toggle pre-check for actions

### PR DESCRIPTION
**Pull request by paulsimonandrews**
https://eaflood.atlassian.net/browse/FCEP-87

Allow pre-check to be toggled on and off for the actions

### CODE REVIEW

File: server/models/actionsModel.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ function actionsModel (actions, errorMessage) {
     return {
       heading: `${action.id}: ${action.description}`,
       actionID: action.id,
+      preCheck: action['pre-check'],
```
Comment (Matthew-Collins-Defra) [inline server/models/actionsModel.js]: In the interests of simplicity, I've renamed the property `precheck` in the backend process and data.

File: server/services/togglePreCheckService.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const config = require('../config')
+const wreck = require('@hapi/wreck').defaults({
+  timeout: config.restClientTimeoutMillis
+})
+
+async function getPreCheck (actionID) {
+  const response = await wreck.get(`${config.actionsUrl}/actions`, { json: true })
+  const actions = response.payload.actions || []
+  const action = actions.find(action => action.id === actionID)
+  return action ? action['pre-check'] : null
```
Comment (Matthew-Collins-Defra) [inline server/services/togglePreCheckService.js]: `action.precheck`

File: test/services/togglePreCheckService.test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+const actionID = 'FG1'
+const enabled = true
+const mockGetActionsResult = {
+  actions: [
+    {
+      id: actionID,
+      description: 'Test',
+      'pre-check': enabled,
```
Comment (Matthew-Collins-Defra) [inline test/services/togglePreCheckService.test.js]: `precheck`

File: server/routes/toggle-pre-check.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  },
+  {
+    method: 'POST',
+    path: '/toggle-pre-check',
+    handler: async function (request, h) {
+      const { actionID, enabled, update } = request.payload
+
+      if (update) {
+        console.log(`Request to set pre-check for actionID=${actionID} to ${enabled}`)
+        // No endpoint to toggle rule yet, but will go in here
```
Comment (Matthew-Collins-Defra) [inline server/routes/toggle-pre-check.js]: Endpoint work is in review, so it's available at /actions/{actionID}

**Review by Matthew-Collins-Defra — CHANGES_REQUESTED** [review]
We shouldn't have awkward characters in code, so I've changed `pre-check` to `precheck` in the data model.
Endpoint is now available (in the PR)

File: server/models/actionsModel.js (hunk 1 above)
Comment (paulsimonandrews) [reply server/models/actionsModel.js]: Done

File: server/services/togglePreCheckService.js (hunk 2 above)
Comment (paulsimonandrews) [reply server/services/togglePreCheckService.js]: Done

File: test/services/togglePreCheckService.test.js (hunk 3 above)
Comment (paulsimonandrews) [reply test/services/togglePreCheckService.test.js]: Done

File: server/routes/toggle-pre-check.js (hunk 4 above)
Comment (paulsimonandrews) [reply server/routes/toggle-pre-check.js]: I think doing this is the scope of the next ticket https://eaflood.atlassian.net/browse/FCEP-102 (which I'll probably pick up anyway). I can either do all that work in this PR, or raise another.

File: test/routes/togglePreCheck.test.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    expect(postResponse.headers.location).toBe('/actions')
+  })
+
+  afterEach(async () => {
+    await server.stop()
+    jest.clearAllMocks()
+  })
+
+  afterAll(() => {
+    jest.unmock('../../server/services/toggleRuleService')
```
Comment (Matthew-Collins-Defra) [inline test/routes/togglePreCheck.test.js]: The unmock call is not needed and shouldn't be used, as it tells jest not to mock the module.

File: test/services/togglePreCheckService.test.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  })
+
+  test('togglePreCheck returns undefined on unsuccessful update', async () => {
+    putPayload = undefined
+    const result = await togglePreCheckService.togglePreCheck('badID', false)
+    expect(result).toBeUndefined()
+  })
+
+  afterAll(() => {
+    jest.unmock('@hapi/wreck')
```
Comment (Matthew-Collins-Defra) [inline test/services/togglePreCheckService.test.js]: The unmock call is not needed and shouldn't be used, as it tells jest not to mock the module.

**Review by Matthew-Collins-Defra — CHANGES_REQUESTED** [review]
Minor comment about jest.unmock, see Phil's comment here. https://github.com/DEFRA/ffc-ce-payment-orchestrator/pull/29#discussion_r372902678
Looks good!

File: test/routes/togglePreCheck.test.js (hunk 5 above)
Comment (paulsimonandrews) [reply test/routes/togglePreCheck.test.js]: Done.

File: test/services/togglePreCheckService.test.js (hunk 6 above)
Comment (paulsimonandrews) [reply test/services/togglePreCheckService.test.js]: Done.

**Review by Matthew-Collins-Defra — APPROVED** [review]
LGTM!

**Pull request merged**